source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -553,7 +574,11 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool end_st
   return status;
 }
 
+// TODO(#36970): Restructure the code. Adding methods like handleDataBuffered(),
+// handleDataStreamed(), handleDataBufferedPartial() to handle each case.
```

Comment:
crazy idea; could you do a small PR first, that adds this structure to the code, and then merge it back into this one?

This is really hard to follow. This function is way too long for humans to understand, making it difficult to review.

Reply:
done